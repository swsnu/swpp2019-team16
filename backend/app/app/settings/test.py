from .base import *
import environ
from redis import ConnectionPool

env = environ.Env()
env.read_env('env/.env.development')

DEBUG = True

# Database
# https://docs.djangoproject.com/en/2.1/ref/settings/#databases

DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.sqlite3',
        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
    }
}

REDIS_CONNECTION_POOL = ConnectionPool(host=env('REDIS_HOST'), port=6379, db=0)
