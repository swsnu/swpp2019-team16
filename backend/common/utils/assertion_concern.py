class AssertionConcern:

    def assert_not_none(self, val, msg):
        if val is None:
            raise TypeError(msg)
