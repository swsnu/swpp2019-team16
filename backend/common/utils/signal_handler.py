import asyncio
import functools
import signal


def register_signal_handler(loop, shutdown):
    for signame in ('SIGINT', 'SIGTERM'):
        loop.add_signal_handler(
            getattr(signal, signame),
            functools.partial(asyncio.ensure_future,
                              shutdown(signame, loop)))


async def shutdown_process(self, sig, loop):
    print('caught {0}'.format(sig))
    tasks = [task for task in asyncio.Task.all_tasks() if task is not
             asyncio.tasks.Task.current_task()]

    list(map(lambda task: task.cancel(), tasks))
    results = await asyncio.gather(*tasks, return_exceptions=True)

    print('finished awaiting cancelled tasks, results: {0}'
          .format(results))
    loop.stop()
