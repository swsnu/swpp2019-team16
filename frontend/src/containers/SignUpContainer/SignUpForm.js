import React, { useState, useCallback, useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { withRouter } from 'react-router-dom';
import SignUp from '../../components/SignUp/SignUp';
import { register, changeField, initalizeForm } from '../../modules/auth/auth';

function SignUpForm({ history }) {
  const dispatch = useDispatch();
  const form = useSelector(({ auth }) => ({
    form: auth.register,
  }));
  var passwordConfirm = null;

  useEffect(() => {
    dispatch(initalizeForm('register'));
  }, [dispatch]);

  const onChange = useCallback(
    e => {
      const { value, name } = e.target;
      if (e.target.name === 'passwordConfirmation') {
        passwordConfirm = e.target.value;
      } else {
        dispatch(
          changeField({
            form: 'register',
            key: name,
            value,
          }),
        );
      }
    },
    [dispatch],
  );

  const onConfirm = useCallback(() => {
    const { email, password, vehicleInfo } = form;
    console.log(form);

    if (!email) {
      window.alert('Email must not be empty.');
    } else if (!password) {
      window.alert('Password must not be empty.');
    } else if (password !== passwordConfirm) {
      window.alert('Password confirm does not match password.');
    } else {
      dispatch(register({ email, password, vehicleInfo }));
      history.push('/request');
    }
  }, [dispatch, form, history]);

  return <SignUp form={form} onChange={onChange} onClick={onConfirm} />;
}

export default withRouter(SignUpForm);
