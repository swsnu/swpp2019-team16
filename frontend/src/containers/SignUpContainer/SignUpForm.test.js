import React from 'react';
import SignUpForm from './SignUpForm';
import { renderWithRedux } from '../../test/utils';
import { MemoryRouter, Route } from 'react-router-dom';

describe('<SignUpForm />', () => {
  const state = {
    auth: {
      register: {
        email: 'email',
        password: 'password',
        vehicleInfo: 'info',
      },
    },
    passwordConfirmation: 'password',
  };
  it('SHOULD match with snapshot', async () => {
    const { container } = renderWithRedux(
      <MemoryRouter initialEntries={['/signup']}>
        <Route component={SignUpForm} path="/signup" />
      </MemoryRouter>,
      state,
    );
    expect(container).toMatchSnapshot();
  });
});
