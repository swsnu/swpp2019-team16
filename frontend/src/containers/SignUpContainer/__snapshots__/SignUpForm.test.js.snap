// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpForm /> SHOULD match with snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa bDWFJH"
  >
    <h3>
      Join Ya-Ta!
    </h3>
    <h1>
      Create Your Account
    </h1>
    <div
      class="MuiBox-root MuiBox-root-2"
    >
      <h3>
        Email
      </h3>
      <input
        id="email-input"
        name="email"
        type="text"
        value=""
      />
      <h3>
        Password
      </h3>
      <input
        id="password-input"
        name="password"
        type="password"
        value=""
      />
      <h3>
        Password Confirm
      </h3>
      <input
        id="password-confirmation-input"
        name="passwordConfirmation"
        type="password"
        value=""
      />
      <h3>
        Vehicle Info (for Driver users)
      </h3>
      <input
        id="vehicle-info-input"
        name="vehicleInfo"
        type="text"
        value=""
      />
      <h3>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-contained"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Confirm
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </h3>
    </div>
  </div>
</div>
`;
